server/lib/rooms/roomTypes/direct.ts

```diff
@@ -53,7 +61,13 @@ roomCoordinator.add(DirectMessageRoomType, {
 				return undefined;
 			}
 
-			return Subscriptions.findOneByRoomIdAndUserId(room._id, Meteor.userId());
+			const uid = userId || getCurrentUserId();
+			if (uid) {
+				return Subscriptions.findOneByRoomIdAndUserId(room._id, uid);
```

Comment:
from what I can see only `fname` and `name` fields are being used from the returned subscription, so it would be good to only return them from the find:
```suggestion
				return Subscriptions.findOneByRoomIdAndUserId(room._id, uid, { fields: { name: 1, fname: 1 } });
```